### PR TITLE
Add order filters

### PR DESCRIPTION
**Pull request by jakirkham**
Partially addresses issue ( https://github.com/jakirkham/dask-ndfilters/issues/13 ).

Creates a module and tests for order-based filters (e.g. minimum, maximum, median, etc.) for use with Dask.

### CODE REVIEW

File: dask_ndfilters/order.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+def _get_origin(footprint, origin=0):
+    size = numpy.array(footprint.shape)
+    ndim = footprint.ndim
+
+    if isinstance(origin, numbers.Real):
+        origin = ndim * (origin,)
+
+    origin = numpy.array(origin)
+    origin = numpy.fix(origin).astype(int)
```
Comment (jakirkham) [inline dask_ndfilters/order.py]: Not sure whether this is an ok thing to do or whether we should just be erroring out on non-integer values. This is what SciPy ndimage's `median_filter` does, but the `minimum_filter` and `maximum_filter` actually error out for not integer values. Asked SciPy for some clarification on this.

xref: https://github.com/scipy/scipy/issues/7343

**Comment by jakirkham** [comment]
Would be curious to know if you have any thoughts on this, @mrocklin.

**Comment by mrocklin** [comment]
Is there anything in particular that you would like me to comment on?  Fro a cursory glance it looks like you were able to compose the scipy functions with `da.map_overlap` decently happily.

**Comment by jakirkham** [comment]
Anything that comes to mind really.

I tried to approach this in a similar way as was done with `dask.array.fft`. Potential concerns could be the degree of magic and whether it is too much. Another concern could be whether the test coverage is really good enough. Are there ways it could be consolidated? Also I found an oddity between some of the SciPy functions in terms of how they handled the `origin` argument. Do you have any thoughts on that point?

**Comment by mrocklin** [comment]
Honestly I've been sufficiently busy that I haven't given this problem enough time to have developed  informed opinions about it.  

We might consider reaching out to some other people to get their thoughts.  @jni comes to mind as potentially interested in this sort of thing.

File: dask_ndfilters/order.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+    size = numpy.array(footprint.shape)
+    half_size = size // 2
+    depth = half_size + abs(origin)
+    depth = tuple(depth)
+
+    # Workaround for a bug in Dask with 0 depth.
+    #
+    # ref: https://github.com/dask/dask/issues/2258
+    #
```
Comment (jakirkham) [inline dask_ndfilters/order.py:86]: xref: https://github.com/dask/dask/issues/2258

Comment (jakirkham) [reply dask_ndfilters/order.py:86]: Should be fixed by PR ( https://github.com/dask/dask/pull/2259 ), which has already been merged.

File: dask_ndfilters/order.py (hunk 1 above)
Comment (jakirkham) [reply dask_ndfilters/order.py]: Probably the easiest thing is to turn this into an error outright. Can always soften it later if we decide to.

Comment (jakirkham) [reply dask_ndfilters/order.py]: It is now a `TypeError` to use anything other than integral value(s) for the `origin`.

**Comment by jakirkham** [comment]
I guess if there are no concerns I'll just merge this. That said, this is clearly the very beginning of this library. So there is still plenty of time to discuss changes and improvements to this and later functionality.

**Pull request merged**